staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -187,8 +187,10 @@ type udsHTTPConnectConnector struct {
 }
 
 func (u *udsHTTPConnectConnector) connect(ctx context.Context) (proxier, error) {
+	udsDialCtx, cancel := context.WithTimeout(ctx, time.Duration(5*time.Second))
+	defer cancel()
 	var d net.Dialer
```

Comment:
why don't you add the timeout directly to the net.Dialer?

https://pkg.go.dev/net#Dialer

```
d := net.Dialer{Timeout: 5*time.Second}
```

it looks much cleaner than adding a context on top

Reply:
Agreed, thanks! I'll amend.

And it looks like net.DialContext will do the right thing:
https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/net/dial.go;l=113-127

Reply:
I went a different way and don't need either; PTAL.